### DISCUSSION TITLE
Error: Section '[Sphinx]' does not have an associated "Action" after Updating MOOSE

### DISCUSSION THREAD

**Discussion by hityyds** [idaholab/moose#29545] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello everyone,

Yesterday I updated MOOSE to the latest version (it had been several months since my last update). After recompiling my custom application successfully, I encountered the following error when running my input file:
```
test.i:79.1: section '[Sphinx]' does not have an associated "Action".
 Common causes:
- you misspelled the Action/section name
- the app you are running does not support this Action/syntax
```
The issue is that I do have this custom action implemented in my application, and it was working fine before the MOOSE update.
```
registerMooseAction("SphinxApp", SphinxAction, "add_variable");
registerMooseAction("SphinxApp", SphinxAction, "add_kernel");
registerMooseAction("SphinxApp", SphinxAction, "add_bc");
registerMooseAction("SphinxApp", SphinxAction, "add_aux_variable");
registerMooseAction("SphinxApp", SphinxAction, "add_aux_kernel");
registerMooseAction("SphinxApp", SphinxAction, "add_user_object");
```
Is there any recent change in MOOSE’s handling of custom actions or input file parsing that I should be aware of? Or could there be additional steps needed after updating MOOSE to ensure custom actions are recognized?

Any guidance would be greatly appreciated.

Thank you!

**Comment by GiudGiud** [comment, marked as answer]
Hello

No we have not changed neither of these.
Do you have in SphixApp.C a call to register SphinxAction to the `[Sphinx]` block?

Guillaume

**Reply by hityyds** [reply]
Thank you for your response. After the update, I forgot to modify the autogenerated SphixApp.C. As you pointed out, I indeed didn’t register SphinxAction in SphixApp.C. After making the necessary changes, the issue has been resolved.

Thanks again for your help!
